docs/src/main/sphinx/sql/match-recognize.rst

```diff
@@ -0,0 +1,575 @@
+===============
+MATCH_RECOGNIZE
+===============
+
+Synopsis
+--------
+
+.. code-block:: text
+
+    MATCH_RECOGNIZE (
+      [ PARTITION BY column [, ...] ]
+      [ ORDER BY column [, ...] ]
+      [ MEASURES measure_definition [, ...] ]
+      [ rows_per_match ]
+      [ AFTER MATCH skip_to ]
+      PATTERN ( row_pattern )
+      [ SUBSET subset_definition [, ...] ]
+      DEFINE variable_definition [, ...]
+      )
+
+Description
+-----------
+
+The ``MATCH_RECOGNIZE`` clause is used to detect patterns across rows of a table. Patterns of interest are
```

Comment:
I suggest to make this a bit less formal:

You can use use regular expressions as row pattern syntax to specify the pattern to use for matching the rows.

Reply:
Can we add a few example use case description here?

Reply:
I said "based on regular expressions" to make it clear that it is not quite a regular expression syntax. I don't want to miss this distinction. 

I'll try to think of some real use cases.